index.js

```diff
@@ -157,18 +180,21 @@ const InProgressProgressReport = {
 // helpers for the progress intent
 
 function handleStateInit() {
-    speechResponse = 'What is your project name?';            
+    speechResponse = 'What is your project name?';
     currentState = STATE_PROJECT_NAME;
 }
 
+// Special case because we do not ask for more than one bullet for name
 function handleStateProjectName(content) {
-    data.projectName = content;
-    speechResponse = 'What are your task plans for the next fifteen days?';            
-    currentState = STATE_PLAN_SHORT;
+    if (content !== undefined) {
+        data.projectName = content;
+        speechResponse = 'What are your task plans for the next fifteen days?';
+        currentState = STATE_PLAN_SHORT;
+    }
 }
 
 function handleStatePlanShort(content) {
-    if (content.includes(TERMINATION_WORD)) {
+    if (containsTerminationWord(content)) {
         content = content.substring(0, content.indexOf(TERMINATION_WORD));
         data.taskPlan15Days.push(content);
```

Comment:
we should make a generalized function. Something like `appendData(key, content) { data[key].push(content) }` then we pass in the key (ie `taskPlan15Days`) and we pass in the content (whatever it is).
The advantage of doing it this way is we can wrap all our `if (content !== undefined)` blocks into there, which saves lines of code.